### PR TITLE
feat(wren-ai-service): multi-llm deployment and setting loading mechanism

### PR DESCRIPTION
**Pull request by paopa**
This PR introduces a new hierarchical configuration system for the Wren AI Service, significantly improving flexibility and maintainability. It also refactors various components to use this new system, enhancing overall code organization and clarity.

### 1. New Settings Loading Mechanism

Implemented a hierarchical settings loading mechanism in `src/config.py`:

- Introduced a `Settings` class using `pydantic_settings.BaseSettings`
- Settings are now loaded in the following order of precedence:
  1. Default values (defined as class attributes)
  2. Environment variables
  3. `.env.dev` file
  4. `config.yaml` file

This approach allows for flexible configuration across different environments while maintaining security for sensitive information.

### 2. Configuration File Processing

- Added support for a `config.yaml` file to configure components, pipelines, and service settings
- Implemented processors for different configuration types (LLM, Embedder, Document Store, Engine, Pipeline) in `src/providers/__init__.py`

### 3. Refactored Provider Initialization

- Updated `generate_components` function to use the new configuration system
- Marked `init_providers` and `Wrapper` class as deprecated

### 4. Environment Variable Consolidation

- Consolidated OpenAI API keys into a single `OPENAI_API_KEY` environment variable
- Similarly consolidated Azure OpenAI API keys into `AZURE_OPENAI_API_KEY`

### 5. Updated Service Initialization

- Modified `src/__main__.py` to use the new `Settings` class for service configuration
- Removed direct environment variable access in favor of the new settings system

### 6. Docker and Deployment Updates

- Updated Docker-related files to reflect the new configuration approach
- Modified Kubernetes deployment files to use the new configuration system
- Modified wren launcher to use the new configuration system

## Benefits

1. **Improved Flexibility**: Easier to configure the service for different environments
2. **Enhanced Security**: Better separation of sensitive information from code
3. **Cleaner Code**: Reduced duplication and improved organization of configuration-related code
4. **Easier Maintenance**: Centralized configuration management makes updates and troubleshooting easier

## Testing

To ensure the functionality of these changes, please conduct testing in the following environments:

### Kubernetes (K8s) Testing

Follow the instructions provided in the deployment README to set up and test in a Kubernetes environment.

### Wren Launcher Testing

1. Modify the following codebase to fetch the compose file and setting file in docker.go
   ```go
	const (
		// please change the version when the version is updated
		WREN_PRODUCT_VERSION    string = "0.10.0"
		DOCKER_COMPOSE_YAML_URL string = "https://raw.githubusercontent.com/Canner/WrenAI/c524a9695b66d909d718f02d4bdc694eb891174b/docker/docker-compose.yaml"
		DOCKER_COMPOSE_ENV_URL  string = "https://raw.githubusercontent.com/Canner/WrenAI/c524a9695b66d909d718f02d4bdc694eb891174b/docker/.env.example"
		AI_SERVICE_CONFIG_URL   string = "https://raw.githubusercontent.com/Canner/WrenAI/c524a9695b66d909d718f02d4bdc694eb891174b/docker/config.example.yaml"
	)
   ```
2. Build and execute the launcher

### Docker Compose Testing

1. Build the updated bootstrap image locally:
   ```
   docker build -t bootstrap:local -f ./docker/bootstrap/Dockerfile ./docker/bootstrap/
   ```
3. Update the docker-compose file to use the locally built bootstrap image.
4. Follow the instructions provided in the docker README to start via Docker Compose
5. Proceed with testing using Docker Compose.

### Local Testing

1. Refer to the project README for local setup instructions.
2. Configure the necessary config files.
3. Use the `just` command to start the service and run tests.

Ensure thorough testing across all environments to validate the changes in the relationship recommendation pipeline.

## Documentation

- Updated `README.md` with new setup instructions
- Added `configuration.md` to explain the new configuration system in detail

This PR represents a significant improvement in how we manage configuration for the Wren AI Service, providing a more robust and flexible foundation for future development.

## Next Step

we have to update  to date the [document](https://docs.getwren.ai/oss/installation/custom_llm), after this PR was be merged

### CODE REVIEW

File: .github/workflows/ai-service-test.yaml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
       - name: Install Just
         uses: extractions/setup-just@v2
         with:
-          just-version: "1.31.0"
+          just-version: "1.36.0"
```
Comment (cyyeh) [inline .github/workflows/ai-service-test.yaml:49]: curious to ask, is there any reason we need to upgrade the just version?

File: docker/.env.example [hunk 2]
```diff
@@ -13,11 +13,14 @@ IBIS_SERVER_PORT=8000
...
 WREN_UI_ENDPOINT=http://docker.for.mac.localhost:3000
 
-# LLM
-LLM_OPENAI_API_KEY=
-EMBEDDER_OPENAI_API_KEY=
-# gpt-4o-mini, gpt-4o, gpt-4-turbo, gpt-3.5-turbo
-GENERATION_MODEL=gpt-4o-mini
+# ai service settings
+QDRANT_HOST=qdrant
+SHOULD_FORCE_DEPLOY=0
```
Comment (cyyeh) [inline docker/.env.example:18]: shoud be 1

File: wren-ai-service/README.md [hunk 3]
```diff
@@ -6,26 +6,89 @@ Please read the [documentation](https://docs.getwren.ai/oss/concept/wren_ai_serv
...
+1. **Install Dependencies**:
+
+   ```bash
+   poetry install
+   ```
+
+2. **Generate Configuration Files**:
+
+   ```bash
+   just init true
```
Comment (cyyeh) [inline wren-ai-service/README.md:34]: I suggest default should be true, since the command is used for local development

**Review by cyyeh — COMMENTED** [review]
Great work! There are some comments below for my first review

**Review by cyyeh — COMMENTED** [review]
I've added gpt-4o to config.yaml for kustomization. Also I am curious about the new launcher process if users choose "CUSTOM"

**Comment by paopa** [comment]
> I've added gpt-4o to config.yaml for kustomization. Also I am curious about the new launcher process if users choose "CUSTOM"

the behavior will be the same as before, it check whether the config.yaml exist, if not, will print the message in console to guide user to follow the instruction to download and modify the config.

**Review by cyyeh — APPROVED** [review]
LGTM, thanks

**Pull request merged**